core/tauri/scripts/core.js

```diff
@@ -13,6 +13,13 @@
     })
   }
 
+  window.__TAURI__.convertFileSrc = function convertFileSrc(filePath, protocol = 'asset') {
+    const path = encodeURIComponent(filePath)
+    return navigator.userAgent.includes('Windows') || navigator.userAgent.includes('Android')
+      ? `https://${protocol}.localhost/${path}`
+      : `${protocol}://localhost/${path}`
```

Comment:
Using `Window::with_webview` could make this logic fail if they edit the webview settings on Windows to not include the string Windows, and for macOS/Linux if they add "Windows" or "Android"